types/debounce-promise/index.d.ts

```diff
@@ -10,6 +10,18 @@ declare namespace debounce {
     }
 }
 
+// func is called with an array of array of parameters if accumulate is true
+// Use Array<[arg0, arg1, ..., argN]> as func's first parameter type for correct hints
+declare function debounce<T extends (...args: any[]) => any>(
```

Comment:
`func` in this case can only take one argument.

```suggestion
declare function debounce<T extends (args: any[]) => any>(
```

Reply:
An even better way to write this new overload is:

```ts
declare function debounce<T extends any[], R>(
    func: (args: [...T][]) => R,
    wait?: number,
    options?: { accumulate: true }
): (
    ...args: T
) => R extends Promise<any>
    ? R
    : Promise<R>;
```

Reply:
Thanks, this is a much better way to do it than sprinkling utility types everywhere like I did! Committed this suggestion, with the small replacement of `[...T][]` with `Array<[...T]>`, since the compiler complained